content/post/debrick-core9g25-using-sam-ba-on-linux.md

```diff
@@ -0,0 +1,286 @@
++++
+title= "Debricking CORE 9G25: Using SAM-BA on Linux"
+date= 2018-05-31T11:02:29+02:00
+series = ["Debrick CORE 9G25"]
+author = "Auke Willem Oosterhoff"
+draft= true
++++
+
+In this post I'll explain how to debrick the CORE 9G25 on a Linux machine. This
+post is the second in a series of series posts. I highly recommend to read the
+[previous post]({{< relref "post/debrick-core9g25-intro.md" >}}) before
+continueing. In that post I explained how the boot process works, how the
+update procedure works and the layout of the NAND. This post will be about
+debricking the NAND using SAM-BA on a Linux machine.
+
+## SAM-BA
+In the previous post I explained that the ROM code of the CPU contains a
+program called SAM Boot Agent (SAM-BA). This program can be used to reflash
+the NAND.
+
+In order to interact with this program in ROM from your Linux machine you need
+a tool. This tool is also called SAM-BA and you can download it from
+[Microchip's][sam-ba] website. For this post I used SAM-BA 3.2.1. If I refer to
+SAM-BA in this post I mean the Linux program unless stated otherwise.
+
+## Applets
+SAM-BA comes with little programs, called applets, to interact with external
+flash or memory. SAM-BA supports different pieces of hardware. The applets
+available depends on the board and you can list them by running `sam-ba -d
+<board> -a help`.
+
+``` bash
+$ ./sam-ba -b sam9xx5-ek -a help
+No port option on command-line, using 'serial'.
+Known applets: lowlevel, extram, sdmmc, serialflash, nandflash
+```
+
+An applet can be loaded using the `-a` flag and most applets will print an help
+text if you run the applet with the `help` parameter. Below you find the
+documentation of the `nandflash` applet.
+
+```bash
+$ ./sam-ba -b sam9xx5-ek -a nandflash:help
+Syntax: nandflash:[<ioset>]:[<bus_width>]:[<header>]
+Parameters:
+    ioset      I/O set
+    bus_width  NAND bus width (8/16)
+    header     NAND header value
+Examples:
+    nandflash                 use default board settings
+    nandflash:2:8:0xc0098da5  use fully custom settings (IOSET2, 8-bit bus, header is 0xc0098da5)
+    nandflash:::0xc0098da5    use default board settings but force header to 0xc0098da5
+For information on NAND header values, please refer to SAM9xx5 datasheet section "10.4.4 Detailed Memory Boot Procedures".
+```
+
+## Commands
+Most applets can be used in conjunction with zero or more commands using the
+`-c` flag. To read the first 1024 (or 0x400 hexadecimal) bytes of the NAND, for
+example, you can use the `read` command like below.
+
+``` bash
+$ ./sam-ba -p serial -b sam9xx5-ek -a nandflash -c read:file.bin:0x0:0x400
+```
```

Comment:
ik neem aan dat je hiermee de eerste 1024 bytes leest. Maar is misschien niet helemaal duidelijk. Dus ik weet niet of je een voorbeeld hebt van wat je dan leest?